src/mesh/mesh_refinement.cpp

```diff
@@ -303,164 +303,33 @@ void MeshRefinement::ProlongateCellCenteredValues(const ParArrayND<Real> &coarse
-          Real dx2p = x2p - x2c;
-          const Real fx2m = coords.x2v(fj);
-          const Real fx2p = coords.x2v(fj + 1);
-          Real dx2fm = x2c - fx2m;
-          Real dx2fp = fx2p - x2c;
-
-          // x1 direction
-          int fi = (i - cib.s) * 2 + ib.s;
-          const Real x1m = coarse_coords.x1v(i - 1);
-          const Real x1c = coarse_coords.x1v(i);
-          const Real x1p = coarse_coords.x1v(i + 1);
-          Real dx1m = x1c - x1m;
-          Real dx1p = x1p - x1c;
-          const Real fx1m = coords.x1v(fi);
-          const Real fx1p = coords.x1v(fi + 1);
-          Real dx1fm = x1c - fx1m;
-          Real dx1fp = fx1p - x1c;
-
-          Real ccval = coarse(n, k, j, i);
-
-          // calculate 2D gradients using the minmod limiter
-          Real gx1m = (ccval - coarse(n, k, j, i - 1)) / dx1m;
-          Real gx1p = (coarse(n, k, j, i + 1) - ccval) / dx1p;
-          Real gx1c =
-              0.5 * (SIGN(gx1m) + SIGN(gx1p)) * std::min(std::abs(gx1m), std::abs(gx1p));
-          Real gx2m = (ccval - coarse(n, k, j - 1, i)) / dx2m;
-          Real gx2p = (coarse(n, k, j + 1, i) - ccval) / dx2p;
-          Real gx2c =
-              0.5 * (SIGN(gx2m) + SIGN(gx2p)) * std::min(std::abs(gx2m), std::abs(gx2p));
-
-          // KGF: add the off-centered quantities first to preserve FP symmetry
-          // interpolate onto the finer grid
-          fine(n, fk, fj, fi) = ccval - (gx1c * dx1fm + gx2c * dx2fm);
-          fine(n, fk, fj, fi + 1) = ccval + (gx1c * dx1fp - gx2c * dx2fm);
-          fine(n, fk, fj + 1, fi) = ccval - (gx1c * dx1fm - gx2c * dx2fp);
-          fine(n, fk, fj + 1, fi + 1) = ccval + (gx1c * dx1fp + gx2c * dx2fp);
-        });
-  } else { // 1D
-    int k = ckb.s, fk = kb.s, j = cjb.s, fj = jb.s;
-    pmb->par_for(
-        "ProlongateCellCenteredValues1d", sn, en, si, ei,
-        KOKKOS_LAMBDA(const int n, const int i) {
-          int fi = (i - cib.s) * 2 + ib.s;
-          const Real x1m = coarse_coords.x1v(i - 1);
-          const Real x1c = coarse_coords.x1v(i);
-          const Real x1p = coarse_coords.x1v(i + 1);
-          Real dx1m = x1c - x1m;
-          Real dx1p = x1p - x1c;
-          const Real fx1m = coords.x1v(fi);
-          const Real fx1p = coords.x1v(fi + 1);
-          Real dx1fm = x1c - fx1m;
-          Real dx1fp = fx1p - x1c;
-
-          Real ccval = coarse(n, k, j, i);
-
-          // calculate 1D gradient using the min-mod limiter
-          Real gx1m = (ccval - coarse(n, k, j, i - 1)) / dx1m;
-          Real gx1p = (coarse(n, k, j, i + 1) - ccval) / dx1p;
-          Real gx1c =
-              0.5 * (SIGN(gx1m) + SIGN(gx1p)) * std::min(std::abs(gx1m), std::abs(gx1p));
-
-          // interpolate on to the finer grid
-          fine(n, fk, fj, fi) = ccval - gx1c * dx1fm;
-          fine(n, fk, fj, fi + 1) = ccval + gx1c * dx1fp;
-        });
+  int b = 0;
+  int nbuffers = coarse.GetDim(6) * coarse.GetDim(5);
+  int nvars = coarse.GetDim(4);
```

Comment:
I don't understand why the different dimensions are distinguished here and only dimensions 5 and 6 are looped over below even though `Get()` appears to return the entire `coarse` and `fine` arrays.

Reply:
I think this is a bug actually. In the original implementation I wrote, there was one `BufferInfo` per dim4 of the pack. But since then, it's been modified to take higher-rank objects. I'll clean this up.

Reply:
Cleaned up.